core/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -206,20 +203,23 @@ private boolean unsubscribe(String queryName,
             return Stream.empty();
         }
 
+        long deadline = System.currentTimeMillis() + unit.toMillis(timeout);
         return handlers.stream()
-                       .map(mh -> {
-                           QueryResponseMessage<R> result = null;
+                       .map(handler -> {
                            try {
-                               result = interceptAndInvoke(DefaultUnitOfWork.startAndGet(interceptedQuery), mh);
+                               long leftTimeout = deadline - System.currentTimeMillis();
```

Comment:
I don't think it's possible, first of all. And since `map` is lazily evaluated, it doesn't mean that terminal operation on stream will invoke all handlers (`findFirst` for example will run the `map` function only once).